server/src/main/java/org/elasticsearch/index/reindex/RemoteInfo.java

```diff
@@ -62,10 +63,16 @@
     private final TimeValue connectTimeout;
 
     public RemoteInfo(String scheme, String host, int port, BytesReference query, String username, String password,
-            Map<String, String> headers, TimeValue socketTimeout, TimeValue connectTimeout) {
+                      Map<String, String> headers, TimeValue socketTimeout, TimeValue connectTimeout) {
+        this(scheme, host, port, null, query, username, password, headers, socketTimeout, connectTimeout);
```

Comment:
Generally I'd prefer not to add method overrides for this unless we have a zillion uses of the old method. Would you be ok with fixing all of the call sites? Maybe in a followup PR?

Reply:
I totally agree with you - there are 11 are call sites - that stopped me to add extra parameter to any of that call site + 10 args... maybe worth to introduce RemoteInfo.Builder for that purpose ?